BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2745,7 +2798,15 @@ void DoRCS()
         {
             try
             {
-                Vector3 relV = TargetVelocity - vessel.Velocity();
+                if (rcsClearanceState == RCSClearanceStates.Clearing || (TimeIndex < dropTime + Mathf.Min(0.5f, BDAMath.SolveTime(10f, currentThrust / part.mass)))) return; // Don't use RCS immediately after launch or when clearing a vessel to avoid running into VLS/SourceVessel
+                Vector3 relV;
+                if (rcsClearanceState == RCSClearanceStates.Turning && SourceVessel) // Clear away from launching vessel
+                {
+                    relV = (part.transform.position - SourceVessel.CoM).normalized + (vessel.Velocity() - SourceVessel.Velocity()).normalized;
```

Comment:
If the missile was directly approaching the source vessel for some reason, then this would give almost zero `relV`.
I'd suggest replacing this with
```csharp
var relP = (part.transform.position - SourceVessel.CoM).normalized;
relV = relP + (vessel.Velocity() - SourceVessel.Velocity()).normalized.ProjectOnPlanePreNormalized(relP);
relV = 100f * relV.ProjectOnPlane(TargetPosition - part.transform.position);
```
so that it's always thrusting away from the source vessel, but with the additional sideways thrusting based on the relative velocity direction. Also, it's using the optimised `ProjectOnPlane` calls.